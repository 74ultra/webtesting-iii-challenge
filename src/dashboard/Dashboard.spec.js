import React from 'react';
import * as rtl from 'react-testing-library';
import Dashboard from './Dashboard.js';

afterEach(rtl.cleanup);

describe('<Dashboard />', () => {
    test('<Dashboard /> snapshot', () => {
        const wrapper = rtl.render(<Dashboard />)
        expect(wrapper.asFragment()).toMatchSnapshot()
    })
    
    
    test('Inital Display defaults to unlocked and open', () => {
        const wrapper = rtl.render(<Dashboard />);
        expect(wrapper.getByText(/unlocked/i))
        expect(wrapper.getByText(/open/i))
        
    })

    test('Intial Controls default to lock gate and close gate', () => {
        const wrapper = rtl.render(<Dashboard />);
        expect(wrapper.getByText(/lock gate/i))
        expect(wrapper.getByText(/close gate/i))
    })


})