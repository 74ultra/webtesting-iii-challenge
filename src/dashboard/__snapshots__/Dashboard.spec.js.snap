// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> <Dashboard /> snapshot 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
      data-testid="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
      data-testid="led green-led"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      data-testid="lockButton"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="closeButton"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;
